extensions/schema-compare/src/test/schemaCompare.test.ts

```diff
@@ -102,12 +93,93 @@ describe('SchemaCompareResult.start', function (): void {
+		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		mockExtensionContext.setup(x => x.extensionPath).returns(() => '');
+		testContext = createContext();
+	});
+
+	beforeEach(async function (): Promise<void> {
+		testContext.apiWrapper.reset();
+	});
+
+	it('Should fail for failing Schema Compare service', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService(testStateScmp.FAILURE);
+
+		testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns((s) => { throw new Error(s); });
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+
+		should(result.getComparisonResult() === undefined);
+
+		result.sourceEndpointInfo = await setEndpointInfo(mocksource);
+		result.targetEndpointInfo = await setEndpointInfo(mocktarget);
+
+		await shouldThrowSpecificError(async () => await result.execute(), loc.compareErrorMessage('Test failure'));
+	});
+
+	it('Should exit for failing Schema Compare service', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService(testStateScmp.FAILURE);
+
+		testContext.apiWrapper.setup(x => x.showErrorMessage(TypeMoq.It.isAny())).returns(() => Promise.resolve(''));
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+
+		should(result.getComparisonResult() === undefined);
+
+		result.sourceEndpointInfo = await setEndpointInfo(mocksource);
+		result.targetEndpointInfo = await setEndpointInfo(mocktarget);
+
+		await result.execute();
+		testContext.apiWrapper.verify(x => x.showErrorMessage(TypeMoq.It.isAny()), TypeMoq.Times.once());
+	});
+
+	it('Should disable script button and apply button for Schema Compare service for dacpac', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService(testStateScmp.SUCCESS_NOT_EQUAL);
+
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+
+		should(result.getComparisonResult() === undefined);
+
+		result.sourceEndpointInfo = await setEndpointInfo(mocksource);
+		result.targetEndpointInfo = await setEndpointInfo(mocktarget);
+
+		await result.execute();
+
+		//Generate script button and apply button should be disabled for dacpac comparison
+		should(result.verifyButtonsState(false, false)).equal(true);
+	});
+
+	it('Should disable script button and apply button for Schema Compare service for database', async function (): Promise<void> {
+		let sc = new SchemaCompareTestService(testStateScmp.SUCCESS_NOT_EQUAL);
+
+		let result = new SchemaCompareMainWindow(testContext.apiWrapper.object, sc, mockExtensionContext.object);
+
+		await result.start(null);
+
+		should(result.getComparisonResult() === undefined);
+
+		result.sourceEndpointInfo = await setEndpointInfo(mocksource);
+		result.targetEndpointInfo = await setEndpointInfo();
```

Comment:
can `setEndpointInfo()` take an enum that specifies if it's creating a dacpac or db  endpoint or be split into something like `setDatabaseEndpointInfo() `and `setDacpacEndpointInfo()` so it's more clear here that the target endpoint is a database?

Reply:
Makes sense. Updated it to split setting the info.